File: src/main/java/imaxct/dao/impl/UserDaoImpl.java
package imaxct.dao.impl;

import imaxct.dao.IUserDao;
import imaxct.domain.User;
import org.springframework.stereotype.Repository;

import java.util.List;


/**
 * Created by imaxct on 17-4-6.
 */

@Repository
public class UserDaoImpl extends BaseDao<User> implements IUserDao {

    public boolean createUser(User user) {
        return this.create(user);
    }

    public User getUserById(int id) {
        return this.find(User.class, id);
    }

    public User getUserByStuNo(String stuNo) {
        return this.uniqueResult("from User where stuNo = ?", stuNo);
    }

    public List<User> getAllUsers() {
        return this.listPage(0, 500, "from User");
    }
}

File: src/main/java/imaxct/dao/impl/UserDaoImpl.kt
package imaxct.dao.impl

import imaxct.dao.IUserDao
import imaxct.domain.User
import org.springframework.stereotype.Repository



@Repository
class UserDaoImpl : BaseDao<User>(), IUserDao {
    override fun clearAllUser(): Int = this.query("delete from User")

    override fun addUsers(list: MutableList<User>): Boolean = this.batchInsert(list)

    override fun updateUser(user: User): Boolean = this.update(user)

    override fun createUser(user: User): Boolean = this.create(user)

    override fun getUserById(id: Int): User = this.find(User::class.java, id)

    override fun getUserByStuNo(stuNo: String): User? = this.uniqueResult("from User where stuNo = ?", stuNo)

    override fun getAllUsers(): List<User> = this.listPage(0, 500, "from User")
}
